src/main/java/com/wiley/autotest/selenium/elements/upgrade/TeasyWindow.java

```diff
@@ -32,6 +35,36 @@ public void switchToLast() {
         driver.switchTo().window(window);
     }
 
+    /**
+     * To be sure that new window appeared after an action.
+     * For example, after clicking on a link with new
+     * window target.
+     *
+     * Example usage: switchToLastAfter(() -> element.click());
+     *
+     * @param action - an action to do before new window appeared
+     *               and before switching.
+     */
+    @Override
+    public void switchToLastAfter(Actions action) {
+        int windowCount = driver.getWindowHandles().size();
+        action.execute();
+        new CustomWaitFor().condition(newWindowAppeared(), windowCount);
+        switchToLast();
+    }
+
+    /**
+     * To be sure that new window appeared after click on element.
+     * For example, after clicking on a link with new
+     * window target.
+     *
+     * @param element - an element to click to new window switch.
+     */
+    @Override
+    public void switchToLastAfter(TeasyElement element) {
```

Comment:
just a note about name being unclear. "switch to last after element" -  says nothing about the click inside.